src/flownet/network_model/_generate_connections.py

```diff
@@ -94,9 +94,46 @@ def _remove_recorded_connections(
     return conn_matrix
 
 
+def _split_additional_flow_nodes(
+    configuration: Any, concave_hull_bounding_boxes: np.ndarray
```

Comment:
I suggest to limit the input to the additional flow nodes; not the entire configuration. This makes the function more "stand alone" and easier to test.

Reply:
Good point, will do!